fluss-server/src/main/java/com/alibaba/fluss/server/kv/KvTablet.java

```diff
@@ -316,30 +312,14 @@ public LogAppendInfo putAsLeader(
                                     //  of the input row are set to null.
                                     BinaryRow newRow = kvRecord.getRow();
                                     walBuilder.append(RowKind.INSERT, newRow);
-                                    appendedRecordCount += 1;
                                     kvPreWriteBuffer.put(
                                             key,
                                             ValueEncoder.encodeValue(schemaId, newRow),
                                             logOffset++);
                                 }
                             }
                         }
-
-                        // if appendedRecordCount is 0, it means there is no record to append, we
-                        // should not append.
-                        if (appendedRecordCount > 0) {
-                            // now, we can build the full log.
-                            return logTablet.appendAsLeader(walBuilder.build());
-                        } else {
-                            return new LogAppendInfo(
-                                    logEndOffsetOfPrevBatch - 1,
-                                    logEndOffsetOfPrevBatch - 1,
-                                    0L,
-                                    0L,
-                                    0,
-                                    0,
-                                    false);
```

Comment:
It's used to fix  the exception `LogSegmentOffsetOverflowException` when append empty log in method `LogSegment#ensureOffsetInRange` since the `largestOffset` will be 0. So I skip to append empty record to avoid that but it will cause replica's writer state out-of-sync.

Let's to see how Kafka solve this....